src/Disks/S3/DiskS3.cpp

```diff
@@ -1201,6 +1197,12 @@ void DiskS3::readRestoreInformation(DiskS3::RestoreInformation & restore_informa
         readText(restore_information.source_path, buffer);
         assertChar('\n', buffer);
 
+        if (!buffer.hasPendingData())
+            return;
+
+        readBoolTextWord(restore_information.detached, buffer);
```

Comment:
Maybe we can use more complex format, like parse string `detached=true`, because it is difficult to remember order and meaning of fields in file?

Reply:
Make sense, will do it.